en/2-wallet/README.md

```diff
@@ -14,3 +14,360 @@
+5. If any leading zeros in the bytes attach 1
+
+So to go from the original private key described above  to the wif format we can use this simple function
+
+```
+// ToWIF converts a NEO private key to a Wallet Import Format string.
+func (priv *PrivateKey) ToWIF() (wif string) {
+	/* See https://en.bitcoin.it/wiki/Wallet_import_format */
+
+	/* Convert the private key to bytes */
+	priv_bytes := priv.ToBytes()
+
+	/* Convert bytes to base-58 check encoded string with version 0x80 */
+	wif = b58checkencode(0x80, priv_bytes)
+
+	return wif
+}
+```
+
+So we can see that WIF is an encoding algorithm of the private key which provides basic error checking, and improved human readability by attaching a version and checksum, as well as encoding it into Base58.
+
+
+## Public key derivation
+Generally cryptocurrencies use the form of cryptography called elliptic curve cryptography. It's used to derive public key from private key and it's computationally infeasible to do the opposite.
+
+Form of elliptic curve equation is the following:  
+*y^2 = x^3 + ax + b*
+
+Bitcoin uses elliptic curve called secp256k1, while NEO uses secp256r1, where k -- means Koblitz and r -- means Random.
+Essentially, secp256k1's parameters were chosen in a way that allows more efficient calculation (for a very small security trade-off), while secp256r1's parameters were chosen randomly.
+
+secp256k1 equation is:  
+*y^2 = x^3 + 7*
+
+secp256r1 equation is:  
+*y^2 = x^3 - 3x + b*, where *b* is *41058363725152142129326129780047268409114441015993725554835256314039467401291*
+
+Because of large *b* used in secp256r1, below we will use secp256k1 for explanation, but principle is the same.
+
+This is how secp256k1 curve looks:  
+![](https://cdn-images-1.medium.com/max/1600/1*4dcCrlQfGqZECDLy-25fnw.png)
+
+First, we'll explain how to do point addition on an elliptic curve.  
+Having points *P* and *Q*, we draw a line that goes through both of them and find third point on the intersection of the curve and the line. Then we reflect that point across x-axis to get *R*.  
+*P + Q = R*  
+![](https://cdn-images-1.medium.com/max/1600/1*dErGh_rL2ExM6AX-Rtyb7w.png)
+
+But in the elliptic cryptography, rather than adding two arbitrary points together, we add specified base point on the curve to itself.  
+We draw tangent line to the curve at the point *P*, then we apply the same rules as above.  
+*P + P = 2 * P*  
+![](https://cdn-images-1.medium.com/max/1600/1*ffYKgW-4_Paxve3G1HIJXw.png)
+
+Remember that private key is a 256-bit number? Basically, public key is the result of *P* adding to itself *x* times, where *x* is our private key.  
+
+*X = x * P*, where *X* is the public key.
+
+To do the opposite (figure out *x* from *X* and *P*) we have to keep adding *P* to itself until we get *X*, which on average make us do *2^128* point additions to figure out *x*, which is computationally infeasible.
+
+## ECDSA signing
```

Comment:
To this section,I think the description in NEO Developer Guide will be better.

Brief steps are as follows:

​　Assume private key, public key and base point as k, K and G, respectively. We know that K = k*G according to ECC algorithm.

Signing procedure:

    Select random number r and compute point r·G(x, y).

    Compute s = (h + k·x)/r according to random number r, message M's hash value h, private key k.

    Send message M and signature {r·G, s} to receiver.

Verification procedure:

    Receiver receives message M and signature {r·G=(x,y), s}.

    Compute hash h according to received message.

    Compute h·G/s + x·K/s with sender public key K and compare with r·G. Verification succeeds if both are the same.

Deduction is as follows:
![image](https://user-images.githubusercontent.com/42929757/57998760-64573300-7b05-11e9-8281-784b212fbb64.png)